datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java

```diff
@@ -46,11 +52,34 @@ public DataSource createDataSource(DataSourceMetadata metadata) {
 
   private void validateGlueDataSourceConfiguration(Map<String, String> dataSourceMetadataConfig)
       throws URISyntaxException, UnknownHostException {
-    DatasourceValidationUtils.validateLengthAndRequiredFields(
-        dataSourceMetadataConfig,
-        Set.of(GLUE_AUTH_TYPE, GLUE_ROLE_ARN, FLINT_URI, FLINT_REGION, FLINT_AUTH));
+    AuthenticationType authenticationType =
+        AuthenticationType.get(dataSourceMetadataConfig.get(GLUE_INDEX_STORE_OPENSEARCH_AUTH));
+    if (AuthenticationType.BASICAUTH.equals(authenticationType)) {
+      DatasourceValidationUtils.validateLengthAndRequiredFields(
+          dataSourceMetadataConfig,
+          Set.of(
+              GLUE_AUTH_TYPE,
+              GLUE_ROLE_ARN,
+              GLUE_INDEX_STORE_OPENSEARCH_URI,
+              GLUE_INDEX_STORE_OPENSEARCH_REGION,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH_USERNAME,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH_PASSWORD));
+    } else if (AuthenticationType.NOAUTH.equals(authenticationType)) {
+      DatasourceValidationUtils.validateLengthAndRequiredFields(
+          dataSourceMetadataConfig,
+          Set.of(
+              GLUE_AUTH_TYPE,
+              GLUE_ROLE_ARN,
```

Comment:
GLUE_ROLE_ARN is not required if NOAUTH?

Reply:
There are two authentication types. one is for glue and other is for index store.
If it is NOAUTH for glue...it will throw exception only IAM_ROLE is allowed.